sklearn/ensemble/_hist_gradient_boosting/grower.py

```diff
@@ -490,6 +509,13 @@ def split_next(self):
         node.right_child = right_child_node
         node.left_child = left_child_node
 
+        left_child_node.weighted_n_node_samples = self.sample_weight[
+            sample_indices_left
+        ].sum()
+        right_child_node.weighted_n_node_samples = self.sample_weight[
+            sample_indices_right
+        ].sum()
+
```

Comment:
Same here.

Reply:
Hmm, I think this section is needed since it computes and stores the `weighted_n_node_samples` for each child node.